Tox toxic

xWNgLWT0mQf opened the pull request:
<!-- Please put the pull request description above -->
Update net-libs/tox and toxic
---

Please check all the boxes that apply:

- [*] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [*] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [*] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [*] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

Chiitoo commented:
Unsure what is going on at my end, and it already happens with the previous version, but it seems like 'toxic' just quits on me after creating a new profile or/and a new password for it.  The config file seems to come out empty as well... I'll need to look more into that.

In any case, 'qa-vdb' suggests:

```
VDB: detected possibly incorrect RDEPEND (net-im/toxic-0.16.0)
 > dev-libs/libconfig:=
 > media-libs/libvpx:=
```

Unsure about libconfig, I see it in BDEPEND already, and looking at other ebuilds, it seems to mostly hang around in DEPEND or RDEPEND... but again, not familiar with it at this time.

xWNgLWT0mQf committed:
net-libs/tox: Drop old versions

This commit removes tox-0.2.13-r1 and tox-0.2.18-r3.
The bug referenced below is related to tox-0.2.18-r3.
It is possible that the same issue applies to later
versions, so I did not close the bug.

Bug: https://bugs.gentoo.org/836707
Signed-off-by:	Josiah Mullins <JoMull01@protonmail.com>

xWNgLWT0mQf committed:
net-libs/tox: Bump to tox-0.2.20

The bug closed below is a request for
this bump to occur.

Closes: https://bugs.gentoo.org/946056
Signed-off-by: Josiah Mullins <JoMull01@protonmail.com>

xWNgLWT0mQf committed:
net-im/toxic: Remove old version

This commit removes toxic-0.11.3-r2

Signed-off-by: Josiah Mullins <JoMull01@protonmail.com>

xWNgLWT0mQf committed:
net-im/toxic: Bump to toxic-0.16.0

The bug below is a request for this bump.

Closes: https://bugs.gentoo.org/947366
Signed-off-by: Josiah Mullins <JoMull01@protonmail.com>

Chiitoo reviewed (commented):
Apologies, it took me a while to get to this again.

I finally got `toxic` to actually run.  It looks like an issue with regards to `ipv6`, or rather lack of it.  I need to run it with `--ipv4`, but that seems like an upstream issue.

As for the actual ebuilds, it looks like `net-im/toxic[video]` links with `vpx`:

```
$ qa-vdb toxic
VDB: detected possibly incorrect RDEPEND (net-im/toxic-0.16.0)
 > media-libs/libvpx:=
 ```

There's also a note on `llvm`, but I suppose that may be correct here:

```
$ qa-vdb toxic
VDB: detected possibly incorrect RDEPEND (net-im/toxic-0.16.0)
llvm-core/llvm:18 <
``` 

Lastly, `pkg_postinst()` could use a database update call:

```
>>> /usr/bin/toxic
 * An example config file has been installed at /usr/share/toxic/toxic.conf.example
 * Feel free to copy it to ~/.config/tox/toxic.conf and modify as desired
 * QA Notice: .desktop files with MimeType= were found installed
 * but desktop mimeinfo cache has not been updated:
 *   /usr/share/applications/toxic.desktop
 * Please make sure to call xdg_desktop_database_update()
 * in pkg_postinst() and pkg_postrm() phases of appropriate pkgs.
>>> net-im/toxic-0.16.0 merged.
```

Thank you!

xWNgLWT0mQf commented:
Thank you for the feedback! Don't worry about the delay. I hope that the latest push resolved the issues present. You may have noticed that toxic is now on version 0.16.1. Sadly, this version depends on features not present on any release of tox. It rather depends on features present in the live version. Therefore I will need to wait to push 0.16.1.

Chiitoo commented on net-im/toxic/toxic-0.16.0.ebuild:
```diff
@@ -40,6 +41,10 @@ RDEPEND="net-libs/tox:=
 		net-libs/tox:=[av]
 	)
+	video? (
+		media-libs/libvpx
```
This could use the slot operator too, also picked up by `qa-vdb` (which you can find via `app-portage/iwdevtools`):

```
$ qa-vdb toxic
VDB: detected possibly incorrect RDEPEND (net-im/toxic-0.16.0)
media-libs/libvpx | media-libs/libvpx:=
```

Chiitoo commented on net-im/toxic/toxic-0.16.0.ebuild:
```diff
@@ -115,3 +122,13 @@ src_install() {
+
+pkg_postrm() {
+	xdg_desktop_database_update
+}
```
I believe the update does not need to be specified here, as it will come by default via `inherit xdg`, but for the `postinst` it is needed since it is specified here (for the `einfo`).

That written, I don't think it hurts to have it either though.

Chiitoo commented:
Thanks!

I edited out a `tab` character from the `net-libs/tox: Drop old versions`, right after `Signed-off-by:`.  It probably wouldn't break anything, but it was probably not intentional there.  :]
